packages/TauSelector/TauSelector.C

```diff
@@ -0,0 +1,141 @@
+///////////////////////////////////////////////////////////////////////////////////////////
+//
+//  Tau Selector: create vectors with selected Taus, veto (or fakeable) Taus...
+//
+//  All SFs and variables are within the Tau definition
+//
+//  To do: definition of veto Taus
+//
+//
+///////////////////////////////////////////////////////////////////////////////////////////
+
+#include "TauSelector.h"
+
+ClassImp(TauSelector);
+TauSelector::TauSelector() : PAFChainItemSelector() {}
+void TauSelector::Summary() {}
+
+void TauSelector::Initialise() {
+  // Initialise TauSelector
+  gIsData       = GetParam<Bool_t>("IsData");
+  gSelection    = GetParam<Int_t>("iSelection");
+  localPath     = GetParam<TString>("WorkingDir");
+
+  // <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+  selTaus 	= std::vector<Lepton>();
+  vetoTaus 	= std::vector<Lepton>();
+  looseTaus = std::vector<Lepton>();
+}
+
+//################################################################
+//## Definition of wps...
+//################################################################
+
+// (Fill if necessary)
+
+
+//################################################################
+//## Tau definitions for each analysis
+//################################################################
+//
+// Use the functions above to define your objects
+//
+// =============================================================== SELECTED TAUS
+Bool_t TauSelector::isGoodTau(Lepton tau){
+  looseLeptons  = GetParam<std::vector<Lepton>>("looseLeptons");
+  if (gSelection == ittHSelec) {
+  	// 	Selection of taus for multilepton ttH analysis:
+  	// idDecayMode = 1, pt > 20, abs(eta) < 2.3, medium wp in idMVA, no one can
+  	// overlap within dR = 0.4 with loose electrons or muons.
+
+  	Bool_t passDecayMode; Bool_t passPt; Bool_t passEta; Bool_t passID;
+  	Bool_t passOverlap; Bool_t passAntiE; Bool_t passAntiMu;
+
+	  passDecayMode	= (idDecayMode == 1);
+	  passPt			= (pt > 20);
+    passEta			= (abs(eta) < 2.3);
+	  passID			= (idMVA >= 3);
+	  passAntiE		= (idAntiE >= 4);
+	  passAntiMu	= (idAntiMu >= 4);
+
+  	passOverlap		= 1;
+  	for (Int_t i = 0; i < looseLeptons.size(); i++) {
```

Comment:
Página 23, líneas 338-342
«Reconstructed τ h candidates are removed if they overlap within ∆R = 0.4 with loose electrons or muons. No dedicated discriminators against background from prompt electrons and muons
are applied since the contribution from background events with additional prompt electrons
and muons passing the τ h selection criteria but not the muon and electron pre-selection requirements is negligible.»